test/test_postponed_annotations.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+import sys
+from pathlib import Path
+
+import pytest
+from dataclasses import dataclass
+from simple_parsing import ArgumentParser, Serializable
+
+from .b import B
+
+
+@pytest.mark.parametrize(
+    'sys_argv, b_value, p_value', [
+        (['test.py', '--v', '1'], 1, None),
+        (['test.py', '--v', '2', '--p', 'test/'], 2, Path('test/')),
+        (['test.py', '--v', '3', '--p', 'test/test1'], 3, Path('test/test1')),
+        pytest.param(['test.py'], None, None, marks=pytest.mark.xfail(reason='no default value in the dataclass'))
+    ]
+)
+def test_postponed_annotations_with_baseclass(sys_argv, b_value, p_value, monkeypatch):
+    monkeypatch.setattr(sys, 'argv', sys_argv)
```

Comment:
Two notes here:
1. You can pass the list of argument arguments to `parse_args` directly. No need to monkeypatch sys.argv here.
2. You might want to use the `TestSetup` class as a base class for class `B`. This way, you could just do
    ```python
    assert B.setup("--v 1 --p test/") == B(v=1, p=Path("test/"))
    ```